Tribler/community/multichain/community.py

```diff
@@ -0,0 +1,427 @@
+        self.logger.info("Crawler: Requesting crawl from node %s, from sequence number %d" %
+                         (base64.encodestring(candidate.get_member().mid).strip(), sequence_number))
+        meta = self.get_meta_message(CRAWL_REQUEST)
+        message = meta.impl(authentication=(self.my_member,),
+                            distribution=(self.claim_global_time(),),
+                            destination=(candidate,),
+                            payload=(sequence_number,))
+        self.dispersy.store_update_forward([message], False, False, True)
+
+    def received_crawl_request(self, messages):
+        for message in messages:
+            self.logger.info("Crawler: Received crawl request from node %s, from sequence number %d" %
+                             (base64.encodestring(message.candidate.get_member().mid).strip(),
+                              message.payload.requested_sequence_number))
+            self.crawl_requested(message.candidate, message.payload.requested_sequence_number)
+
+    def crawl_requested(self, candidate, sequence_number):
+        blocks = self.persistence.get_blocks_since(self._public_key, sequence_number)
+        if len(blocks) > 0:
+            self.logger.info("Crawler: Sending %d blocks" % len(blocks))
+            messages = [self.get_meta_message(CRAWL_RESPONSE)
+                        .impl(authentication=(self.my_member,),
+                              distribution=(self.claim_global_time(),),
+                              destination=(candidate,),
+                              payload=block.to_payload()) for block in blocks]
+            self.dispersy.store_update_forward(messages, False, False, True)
+            if len(blocks) > 1:
+                # we sent more than 1 block. Send a resumption token so the other side knows it should continue crawling
+                # last_block = blocks[-1]
+                # resumption_number = last_block.sequence_number_requster if
+                # last_block.mid_requester == self._mid else last_block.sequence_number_responder
+                message = self.get_meta_message(CRAWL_RESUME).impl(authentication=(self.my_member,),
+                                                                   distribution=(self.claim_global_time(),),
+                                                                   destination=(candidate,),
+                                                                   # payload=(resumption_number))
+                                                                   payload=())
+                self.dispersy.store_update_forward([message], False, False, True)
+        else:
+            # This is slightly worrying since the last block should always be returned.
+            # Or rather, the other side is requesting blocks starting from a point in the future.
+            self.logger.info("Crawler: No blocks")
```

Comment:
The worrying part is why the crawler is requesting this from the crawlee, there is no reason for the crawlee (where this code runs) to worry about this. So I would say that for the expected use cases of this log message, Info is an appropriate level.